storage/Filesystems/MountPointImpl.cc

```diff
@@ -0,0 +1,492 @@
+
+	Device::Impl::log_diff(log, rhs);
+
+	storage::log_diff(log, "path", path, rhs.path);
+
+	storage::log_diff_enum(log, "mount-by", mount_by, rhs.mount_by);
+
+	storage::log_diff(log, "mount-options", mount_options.get_opts(), rhs.mount_options.get_opts());
+
+	storage::log_diff(log, "freq", freq, rhs.freq);
+ 	storage::log_diff(log, "passno", passno, rhs.passno);
+
+	storage::log_diff(log, "active", active, rhs.active);
+ 	storage::log_diff(log, "in-etc-fstab", in_etc_fstab, rhs.in_etc_fstab);
+    }
+
+
+    void
+    MountPoint::Impl::print(std::ostream& out) const
+    {
+	Device::Impl::print(out);
+
+	out << " path:" << path;
+
+	out << " mount-by:" << toString(mount_by);
+
+	out << " mount-options:" << mount_options.get_opts();
+
+	out << " freq:" << freq << " passno:" << passno;
+
+	if (active)
+	    out << " active";
+
+	if (in_etc_fstab)
+	    out << " in-etc-fstab";
+    }
+
+
+    bool
+    MountPoint::Impl::has_mountable() const
+    {
+	return has_single_parent_of_type<const Mountable>();
+    }
+
+
+    Mountable*
+    MountPoint::Impl::get_mountable()
+    {
+	return get_single_parent_of_type<Mountable>();
+    }
+
+
+    const Mountable*
+    MountPoint::Impl::get_mountable() const
+    {
+	return get_single_parent_of_type<const Mountable>();
+    }
+
+
+    string
+    MountPoint::Impl::get_mount_name() const
+    {
+	return get_mountable()->get_impl().get_mount_name();
+    }
+
+
+    string
+    MountPoint::Impl::get_mount_by_name() const
+    {
+	return get_mountable()->get_impl().get_mount_by_name();
+    }
+
+
+    FsType
+    MountPoint::Impl::get_mount_type() const
+    {
+	return get_mountable()->get_impl().get_mount_type();
+    }
+
+
+    void
+    MountPoint::Impl::set_default_mount_by()
+    {
+	set_mount_by(get_mountable()->get_impl().get_default_mount_by());
+    }
+
+
+    void
+    MountPoint::Impl::set_mount_options(const MountOpts& mount_options)
+    {
+	Impl::mount_options = mount_options;
+    }
+
+
+    void
+    MountPoint::Impl::set_mount_options(const vector<string>& mount_options)
+    {
+	Impl::mount_options.set_opts(mount_options);
+    }
+
+
+    void
+    MountPoint::Impl::set_default_mount_options()
+    {
+	set_mount_options(get_mountable()->get_impl().get_default_mount_options());
+    }
+
+
+    void
+    MountPoint::Impl::add_create_actions(Actiongraph::Impl& actiongraph) const
+    {
+	vector<Action::Base*> actions;
+
+	if (active)
+	    actions.push_back(new Action::Mount(get_sid()));
+
+	if (in_etc_fstab)
+	    actions.push_back(new Action::AddToEtcFstab(get_sid()));
+
+	actiongraph.add_chain(actions);
+    }
+
+
+    void
+    MountPoint::Impl::add_modify_actions(Actiongraph::Impl& actiongraph, const Device* lhs_base) const
+    {
+	Device::Impl::add_modify_actions(actiongraph, lhs_base);
+
+	const Impl& lhs = dynamic_cast<const Impl&>(lhs_base->get_impl());
+
+	if (!lhs.active && active)
+	{
+	    Action::Base* action = new Action::Mount(get_sid());
+	    actiongraph.add_vertex(action);
```

Comment:
A vector and using add_chain also adds dependencies between the actions of the vector. So just changing the code here will result in a different actiongraph.